spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/BindableProxyFactory.java

```diff
@@ -274,10 +126,9 @@ public void bindInputs(ChannelBindingService channelBindingService) {
 		if (log.isDebugEnabled()) {
 			log.debug(String.format("Binding inputs for %s:%s", this.channelNamespace, this.type));
 		}
-		for (Map.Entry<String, ChannelHolder> channelHolderEntry : this.inputs.entrySet()) {
+		for (Map.Entry<String, ChannelHolder> channelHolderEntry : this.channelFactory.getInputs().entrySet()) {
```

Comment:
Yeah, that was my mistake to move the channels into `ChannelFactory`. Pushed the fix.